lib/Analysis/Utility.cpp

```diff
@@ -157,14 +157,18 @@ std::string getValueOperandName(Value value, AsmState &state) {
   return opName;
 }
 
-bool isMmaToDotShortcut(triton::gpu::MmaEncodingAttr &mmaLayout,
-                        triton::gpu::DotOperandEncodingAttr &dotOperandLayout) {
+bool isMmaToDotShortcut(RankedTensorType &srcTy, RankedTensorType &dstTy) {
   // dot_op<opIdx=0, parent=#mma> = #mma
   // when #mma = MmaEncoding<version=2, warpsPerCTA=[..., 1]>
+  auto srcLayout = srcTy.getEncoding();
+  auto dstLayout = dstTy.getEncoding();
+  auto mmaLayout = srcLayout.cast<triton::gpu::MmaEncodingAttr>();
+  auto dotOperandLayout = dstLayout.cast<triton::gpu::DotOperandEncodingAttr>();
   return mmaLayout.getVersionMajor() == 2 &&
          mmaLayout.getWarpsPerCTA()[1] == 1 &&
          dotOperandLayout.getOpIdx() == 0 &&
-         dotOperandLayout.getParent() == mmaLayout;
+         dotOperandLayout.getParent() == mmaLayout &&
+         !srcTy.getElementType().isF32();
```

Comment:
Hmm, why would `isMmaToDotShortcut` not work with float32 data?

Reply:
Because the accumulator doesn't match the dot_op layout on "mma.sync.aligned.m16n8k8.row.col.f32.tf32.tf32.f32"